### PR TITLE
ELY-2769 Update CI to also run with SE 21 for branch 2.2.x

### PR DESCRIPTION
**Pull request by PrarthonaPaul**
https://issues.redhat.com/browse/ELY-2769
Depends on issues tracked through https://issues.redhat.com/browse/ELY-2760

Related issue: https://issues.redhat.com/browse/ELY-2727
Related PR: https://github.com/wildfly-security/wildfly-elytron/pull/2103

### CODE REVIEW

File: .github/workflows/pr-ci.yaml [hunk 1]
```diff
@@ -15,12 +15,13 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest, windows-latest, macos-latest]
+        java: ['11', '17', '21']
     steps:
       - uses: actions/checkout@v2
       - name: Set up JDK 11
```
Comment (fjuma) [inline .github/workflows/pr-ci.yaml]: This needs to be updated too (see https://github.com/wildfly-security/wildfly-elytron/pull/2142)

Comment (PrarthonaPaul) [reply .github/workflows/pr-ci.yaml]: fixed, thanks!

Comment (PrarthonaPaul) [reply .github/workflows/pr-ci.yaml]: Hi @fjuma 
It looks like there are some test failures with Java 17 on this branch: https://github.com/wildfly-security/wildfly-elytron/actions/runs/9288164876/job/25558904163?pr=2153#step:6:4956

Comment (fjuma) [reply .github/workflows/pr-ci.yaml]: Can you check if the failures are the same as the ones tracked here:

https://issues.redhat.com/browse/ELY-2760

The issues linked to ELY-2760 will need to be merged first.

Comment (PrarthonaPaul) [reply .github/workflows/pr-ci.yaml]: Yes, it is related to https://issues.redhat.com/browse/ELY-2763. 
I will add a note here that it depends on https://issues.redhat.com/browse/ELY-2760
Thanks!

**Pull request merged**